### ISSUE TITLE
[Feature request] avoid regenerating pages that do not contain changes

### ISSUE THREAD

**Issue by rogervila** [twisted/pydoctor#521]
Hello,

I am generating using git to track changes on my documentation website generated with _pydoctor_.

When I regenerate the HTML output, all files are changed with the current timestamp. For example:

```HTML
  <div class="container">
    <a href="index.html">API Documentation</a> for equipment,
  generated by <a href="https://github.com/twisted/pydoctor/">pydoctor</a>
    22.3.0 at 2022-03-16 12:45:05.
  </div>
```

I would like to have an option to **avoid regenerating pages that do not contain changes** to keep the originally generated timestamp.

This is the command I am using to generate the docs.

```sh
pydoctor --make-html --html-output=website/static/api my-package
```
Thank you.

**Comment by adiroiban** [comment]
Thanks for the report.

PR welcomed :)

**Comment by rogervila** [comment]
Good to know, I will work on it as soon as I have bandwidth!

**Comment by tristanlatr** [comment]
Hi @rogervila ,

Thanks for the feature request

Is it related to the speed of pydoctor ? I agree that pydoctor could be optimized at several levels.

Detect if the generated page will be same as the already present html without flattening the new object page as html seem like a very hard task. There are a lot of extrapolated Informations in the pages, like subclasses, overridden methods, etc. And, for now, a lot of this information is computed at the time of the html generation. So I’d say you’ll have to generate the html anyway. Then you’ll have to compare html files, most probably with regular expression. I think this could be done by generating documentation to a temp folder and synch each file that has changed/added/removed. But this approach will not improve pydoctor performance.

Tell me what you think

**Comment by HaraldKi** [comment]
Yes, I saw the `--buildtime` option, but to me it feels a bit like cheating whoever reads the doc, so I rather stick with a footer into which I can generate a `VERSION: x.y.z` whenever suitable.

Maybe a `--buildversion` option which can override  the `--buildtime` option could be a thing. (Just musing.)

**Comment by HaraldKi** [comment]
Sorry for the late reply. Didn't have credential on a vacation:-) 

#333 combine with something like `--buidltime=none` seems like a perfect combination.
